src/data/shopping_list.ts

```diff
@@ -0,0 +1,34 @@
+import { HomeAssistant } from "../types.js";
+import { ShoppingListItem } from "../panels/lovelace/cards/hui-shopping-list-card.js";
+
+export const fetchItems = (hass: HomeAssistant): Promise<ShoppingListItem[]> =>
+  hass.callApi("GET", "shopping_list");
+
+export const saveEdit = (
+  hass: HomeAssistant,
+  itemId: number,
+  name: string
+): Promise<{}> =>

```

Comment:
`Promise<void>`, as you don't expect any return value.

Reply:
actually, you get the shopping list item back. Did you look at the actual return values on the backend?